src/components/Wallet/WalletGenerator.jsx

```diff
@@ -81,39 +101,185 @@ class WalletGenerator extends React.Component {
     );
   }
 
+  async componentDidUpdate(prevProps) {
+    const prevPassword = prevProps.client.password
+    const { setPasswordError, network, client } = this.props;
+    // if the password was updated
+    if (prevPassword !== client.password && client.password.length) {
+      // test the connection using the set password
+      // but only if the password field hasn't been changed for 500ms
+      this.throttleTestConnection({ network, client, setPasswordError })
+    }
+  }
+
+  async handlePasswordChange(event) {
+    event.preventDefault()
+    const { setPassword, setPasswordError } = this.props;
+    const password = event.target.value;
+    this.setState({ connectSuccess: false }, () => {
+      setPassword(password);
+      setPasswordError('')
+    });
+  };
+
+  async handlePasswordEnter(event) {
+    event.preventDefault()
+    this.throttleTestConnection.cancel()
+    await this.testConnection(this.props, this.generate)
+  }
+
+  /**
+   * Callback function to pass to the address importer
+   * after addresses have been imported we want
+   * @param {Array<string>} importedAddresses 
+   * @param {boolean} rescan - whether or not a rescan is being performed
+   */
+  async afterImportAddresses(importedAddresses, rescan) {
```

Comment:
For the generator, after the addresses have been imported, this requests slice information for all imported addresses.